Automattic/wp-calypso
# Signup: Update DesignTypeWithAtomicStoreStep to use TileGrid

This PR updates `DesignTypeWithAtomicStoreStep` to use the `TileGrid` component family for displaying all the tiles. It also reuses some classes from `DesignTypeWithStoreStep` - we can later move these to a more common location, so they can be reused. 

A change that's worth mentioning is that it will now rely on `this.state.showStore` directly for showing/hiding the atomic store, vs. unnecessarily using a className for that.

This allows to save a bunch of CSS and markup that we're reusing in other design type steps in Signup and Jetpack Onboarding. Can be followed by a couple more cleanup PRs after merging.

Contains #20640, which fixes a small visual bug with card link indicators on mobile devices.

To test:
* Checkout this branch
* Go to http://calypso.localhost:3000/start/design-type-with-store-nux
* Try both different variants of the `signupAtomicStoreVsPressable` AB test.
* Verify there are no visual/functional differences when comparing to production.
* Verify the step works like it did before.
